From faa181804e60bc0e747cb0618cd7db3b605ec767 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 26 Jan 2020 00:40:13 +0100
Subject: [PATCH 042/312] media: sun6i-csi: Fix vidioc_g/s_parm to handle lack
 of the sensor

In case no link is enabled, sun6i_video_remote_subdev would return NULL.
Handle this gracefully.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c b/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
index 870129ce99a8..73c3927afa57 100644
--- a/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
+++ b/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
@@ -446,6 +446,8 @@ static int vidioc_g_parm(struct file *file, void *priv,
 	struct v4l2_subdev *subdev;
 
 	subdev = sun6i_video_remote_subdev(video, NULL);
+	if (!subdev)
+		return -ENXIO;
 
 	return v4l2_g_parm_cap(video_devdata(file), subdev, p);
 }
@@ -457,6 +459,8 @@ static int vidioc_s_parm(struct file *file, void *priv,
 	struct v4l2_subdev *subdev;
 
 	subdev = sun6i_video_remote_subdev(video, NULL);
+	if (!subdev)
+		return -ENXIO;
 
 	return v4l2_s_parm_cap(video_devdata(file), subdev, p);
 }
-- 
2.31.1

